src/oops/SOLID/lsp/stack/before/StackWrong.java

```diff
@@ -14,29 +14,39 @@
  * so objects of ArrayList are not fully replaceable by the objects of stack.   
  * 
  */
-public class StackWrong extends ArrayList<Integer>{
+public class StackWrong {
 	private int topPointer = 0;
-	
+	ArrayList<Integer> stack;
+	public StackWrong() {
+		this.stack = new ArrayList<Integer>();
+	}
 	public void push(Integer a) {
-		add(topPointer, a);
+		stack.add(topPointer, a);
 		topPointer++;
 	}	
 	public void pop() {
-		remove(topPointer-1);
+		stack.remove(topPointer-1);
 		topPointer--;
 	}
 	public Integer top() {
-		return get(topPointer-1);
+		return stack.get(topPointer-1);
 	}
-	
+
 	public static void main(String[] args) {
```

Comment:
We should not expose our method definition to users. You should keep this in a separate file. If client wants to make any changes in your implemented methods of Stack he/she can do it.